hack/update-vendor.sh

```diff
@@ -51,4 +48,48 @@ if [ ${need_update} = true ]; then
   exit 1
 fi
 
+# hack/versions should be correct now.
+echo "Compare vendor with containerd vendors..."
+source hack/versions
+if [ -z "${CONTAINERD_REPO}" ]; then
+  CONTAINERD_REPO=containerd/containerd
+else
+  CONTAINERD_REPO=${CONTAINERD_REPO#*/}
+fi
+containerd_vendor=$(mktemp /tmp/containerd-vendor.conf.XXXX)
+curl -s https://raw.githubusercontent.com/${CONTAINERD_REPO}/${CONTAINERD_VERSION}/vendor.conf > ${containerd_vendor}
+# Create a temporary vendor file to update.
+tmp_vendor=$(mktemp /tmp/vendor.conf.XXXX)
+while read vendor; do
+  repo=$(echo ${vendor} | awk '{print $1}')
+  commit=$(echo ${vendor} | awk '{print $2}')
+  vendor_in_containerd=$(grep ${repo} ${containerd_vendor} || true)
+  if [ -z "${vendor_in_containerd}" ]; then
+    echo ${vendor} >> ${tmp_vendor}
+    continue
+  fi
+  commit_in_containerd=$(echo ${vendor_in_containerd} | awk '{print $2}')
+  if [ "${commit}" != "${commit_in_containerd}" ]; then
+    echo ${vendor_in_containerd} >> ${tmp_vendor}
+  else
+    echo ${vendor} >> ${tmp_vendor}
+  fi
```

Comment:
needs checks for optional github repo link.. For example: see "https://github.com/golang/sys" below

Reply:
If repo is different, I guess commit id will be different. I think commit id is uniquely reflected the commit history.
If the commit id is the same, I think it means the code should also be exactly the same, even they are in different repo. :)

Reply:
I can add a comment here. :)

Reply:
v1.0 type tags are more risky.. also there may be some odd issue of setup. I'd also expect failures but really the issue was more of a why not check that too while checking version in case someone wanted to link to a personal repo fork for a bit.

Reply:
I agree odds are low but there could be a match and still a difference in link. Another example, we or containerd may have a superfluous link that should be removed.

Reply:
Done.